### PR TITLE
FEAT: detailed card completed with correction based on previous issue #1

### PR DESCRIPTION
**Pull request by gami34**
What this PR does (required):
all corrections raised during issue #1 were corrected and implemented
API to add Card's description, deadline, and comment feature has been addressed

Screenshots / Videos (front-end only):
Post at least one screenshot to show the feature in action. Only required in front-end work

Any information needed to test this feature (required):
add a MongoDB URI to the "MONGO_URI" key in an .env file on the servers' folder root directory. A MongoDB Atlas URI will be perfect for this
then add a Secret String Type Data to the "SECRET" key on the .env on the servers' folder root directory.
then run script below on your terminal from the server's folder root directory

```bash
$ npm run dev
```
Kindly find API Documentation for data format purposes
https://documenter.getpostman.com/view/10413337/UV5f6tSz

Any issues with the current functionality (optional):
List any problems where you were not able to complete all tasks on ticket
List any problems this PR may cause for the project or other active PRs
Post a screenshot/video of the problem, along with detailed console outputs where applicable

### CODE REVIEW

File: server/models/board.js [hunk 1]
```diff
@@ -1,15 +1,103 @@
...
+
+  if (boards.length == 0) {
+    let newBoard = await new this({
+      userId: userId,
+      name: 'my board',
+    })
+    let columns = await this.model('Column').generateBoardColumns(newBoard._id)
+    newBoard.columns = columns
+    await newBoard.save()
+    // refetch the board data n populate
```
Comment (ezeilo-su) [inline server/models/board.js:65]: Remove comment

File: server/models/board.js [hunk 2]
```diff
@@ -1,15 +1,103 @@
...
+  }
+  return boards
+}
+
+boardSchema.statics.updateBoardColumnsList = async function (
+  boardId,
+  columnId,
+  index
+) {
+  //  pop the columnId if it exist
```
Comment (ezeilo-su) [inline server/models/board.js:79]: Remove comment

File: server/models/board.js [hunk 3]
```diff
@@ -1,15 +1,103 @@
...
+  await this.findOneAndUpdate(
+    { _id: boardId, columns: columnId },
+    {
+      $pullAll: { columns: [columnId] },
+    }
+  )
+
+  // update the user board
```
Comment (ezeilo-su) [inline server/models/board.js:87]: This one too

File: server/models/card.js [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+    collection: 'cards',
+    timestamps: true,
+  }
+)
+
+cardSchema.statics.createColumnCard = async function (
+  { name, description, index },
+  columnId
+) {
+  // create the column
```
Comment (ezeilo-su) [inline server/models/card.js:23]: Let get rid of this comment

File: server/models/card.js [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+  let card = await new this({
+    name: name,
+    description: description || '',
+    columnId: columnId,
+  })
+  await card.save()
+  // update the Column Card List board
```
Comment (ezeilo-su) [inline server/models/card.js:30]: Get rid of this one

File: server/models/card.js [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+        description: description || previousCardData.description,
+        deadline: deadline || previousCardData.deadline,
+        comment: comment || previousCardData.comment,
+        description: description || previousCardData.description,
+      },
+    }
+  )
+
+  if (index >= 0) {
+    // update the user board
```
Comment (ezeilo-su) [inline server/models/card.js:60]: Remove comment

File: server/models/column.js [hunk 7]
```diff
@@ -1,15 +1,156 @@
...
+    })
+  }
+  return columnsObjectIds
+}
+
+columnSchema.statics.createBoardColumn = async function (
+  { name, index },
+  boardId
+) {
+  // create the column
```
Comment (ezeilo-su) [inline server/models/column.js:43]: Remove comment

File: server/models/column.js [hunk 8]
```diff
@@ -1,15 +1,156 @@
...
+  let column = await new this({
+    name: name,
+    boardId: boardId,
+    cards: [],
+  })
+  await column.save()
+  // update the user board
```
Comment (ezeilo-su) [inline server/models/column.js:51]: Remove

File: server/models/column.js [hunk 9]
```diff
@@ -1,15 +1,156 @@
...
+      { _id: columnId },
+      {
+        $set: {
+          name: name,
+        },
+      }
+    )
+  }
+  if (index >= 0) {
+    // update the user board
```
Comment (ezeilo-su) [inline server/models/column.js:72]: Remove comment

File: server/models/column.js [hunk 10]
```diff
@@ -1,15 +1,156 @@
...
+  )
+  await this.deleteMany({ _Id: columnId })
+  await this.model('Card').deleteMany({ columnId: columnId })
+}
+
+columnSchema.statics.updateColumnCardsList = async function (
+  { cardId, index },
+  columnId
+) {
+  //  pop the cardId if it exist
```
Comment (ezeilo-su) [inline server/models/column.js:94]: Also remove this one

File: server/models/column.js [hunk 11]
```diff
@@ -1,15 +1,156 @@
...
+  await this.findOneAndUpdate(
+    { _id: columnId },
+    {
+      $pullAll: { cards: [cardId] },
+    }
+  )
+
+  // update the Column with new cardId
```
Comment (ezeilo-su) [inline server/models/column.js:102]: This one too

File: server/models/column.js [hunk 12]
```diff
@@ -1,15 +1,156 @@
...
+    }
+  )
+}
+
+columnSchema.statics.updateTargetColumnCardsList = async function (
+  { cardId, index },
+  currentColumnId,
+  targetColumnId
+) {
+  //  pop the cardId if it exist
```
Comment (ezeilo-su) [inline server/models/column.js:121]: Remove comment

File: server/models/column.js [hunk 13]
```diff
@@ -1,15 +1,156 @@
...
+  await this.findOneAndUpdate(
+    { _id: currentColumnId },
+    {
+      $pullAll: { cards: [cardId], cardsSort: [cardId] },
+    }
+  )
+
+  // update the Column with new cardId
```
Comment (ezeilo-su) [inline server/models/column.js:129]: Remove comment

File: server/models/column.js [hunk 14]
```diff
@@ -1,15 +1,156 @@
...
+        },
+        cardsSort: {
+          $each: [cardId],
+          $position: index,
+        },
+      },
+    }
+  )
+
+  // update the card with its new targetId
```
Comment (ezeilo-su) [inline server/models/column.js:146]: Remove comment

**Review by ezeilo-su — CHANGES_REQUESTED** [review]
Quite a good work on this one :+1:
#### I have some comments for you
- So many file changes occurred due to your linters setup which is not consistent with the provided style guide. Please address this and make sure to apply proper formatting on your code, as I could not comment on each line of code in all cases.
- Your code is well written and nicely documented. Please remove unnecessary comments and console logs.
Please make the requested changes and wait for approval by a team member.
Happy coding!

**Comment by tejasj1718** [comment]
Hey Gideon,

I think you must merge this PR without waiting for any approval and merge the original issue. If you added another new issue into this PR then do not do that. You must open a new PR into main for that. Please clarify this with Ethan as well!

**Comment by gami34** [comment]
Dear Team,

All comments and console logs have been taken out.

we are still maintaining the same Linting

File: server/models/board.js [hunk 15]
```diff
@@ -1,15 +1,103 @@
...
+    )
+  }
+  const board = await this.findOne({ _id: boardId })
+  return board
+}
+
+boardSchema.statics.deleteBoardBoard = async function (boardId) {
+  const board = await this.findOne({ _id: boardId })
+  await board?.columns.forEach((columnObjectId) => {
+    this.model('Column').deleteMany({ _Id: columnObjectId })
```
[the remark this answers is not in the document]
Comment (gami34) [reply server/models/board.js:44]: thanks allot for this

**Review by moffatethan — APPROVED** [review]
@gami34 Good changes! 🚀

**Pull request merged**